src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbFilter/breadcrumbFilter.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import {t} from 'app/locale';
+import DropdownControl from 'app/components/dropdownControl';
+import DropdownButton from 'app/components/dropdownButton';
+
+import BreadcrumbFilterGroup from './breadcrumbFilterGroup';
+import BreadcrumbFilterHeader from './breadcrumbFilterHeader';
+import BreadcrumbFilterFooter from './breadcrumbFilterFooter';
+import {FilterGroup, FilterGroupType, FilterType} from './types';
+
+type Props = {
+  onFilter: (filterGroups: Array<FilterGroup>) => () => void;
+  filterGroups: Array<FilterGroup>;
+};
+
+type State = {
+  filterGroups: Array<FilterGroup>;
+};
+
+class BreadcrumbFilter extends React.Component<Props, State> {
+  state: State = {
+    filterGroups: [],
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(this.props.filterGroups, prevProps.filterGroups)) {
+      this.loadState();
```

Comment:
Do you need to do this on mount as well?

Reply:
yes.. because of async issues this component get's the necessary props later on...